spec/lib/audit/catalog_to_moab_class_spec.rb

```diff
@@ -6,31 +6,41 @@
   let(:storage_dir) { 'spec/fixtures/storage_root01/moab_storage_trunk' }
   let(:limit) { Settings.c2m_sql_limit }
 
-  context '.check_version_on_dir_of_batch' do
+  context '.check_version_on_dir' do
     include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit) }
+    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit) }
 
-    it 'creates an instance and calls #check_catalog_version' do
-      c2m_mock = instance_double(described_class)
-      allow(described_class).to receive(:new).and_return(c2m_mock)
-      expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
-      described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit)
-    end
-  end
+    context 'when there are PreservedCopies to check' do
+      let(:c2m_mock) { instance_double(described_class) }
 
-  context ".check_version_on_dir" do
-    include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir) }
+      it 'creates an instance and calls #check_catalog_version within the limit' do
+        allow(described_class).to receive(:new).and_return(c2m_mock)
+        expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
+        described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit)
+      end
 
-    it "calls check_version_on_dir_of_batch when there are objects to audit" do
-      expect(described_class).to receive(:check_version_on_dir_of_batch).exactly(1).times
-      subject
+      it 'creates an instance and calls #check_catalog_version on everything in batches' do
+        # we use 2 separate loops as we have to do all of our .new calls on CatalogToMoab before the allowance
```

Comment:
what's "the allowance" ?  I wonder if this comment can be deleted.   Basically, what I see is "here's a bunch of setup, with allows and expects, and here's the call to the method.   That's pretty standard rspec fare, to me.  I'm open to other opinions.

Reply:
the specific tricky part is that the allocation of real `CatalogToMoab` objects has to all happen up front in the first loop, because otherwise you set an `allow` on `C2M#new` for the first of the c2m test instances, but then immediately call `C2M#new` with parameters for the next c2m instance, and then rspec complains that you called `#new` with unexpected params.  that bit us and took us a few minutes to figure out.  our first inclination was to put everything in one loop.  the intent was just to explain why the intuitive thing doesn't work.

Reply:
I'm in agreement with @jmartin-sul -- the setup was unintuitive, and the comment is hopefully informative in that light.

Reply:
ah - ok.  I'm struggling with the word "allowance" - i don't connect it with "allow" statements.  How about

```ruby
# we must set up all the described_class instance objects ahead of any process calling CatalogToMoab.new
```